test/unit/presenters/document_export_presenter_test.rb

```diff
@@ -49,9 +49,32 @@ class DocumentExportPresenterTest < ActiveSupport::TestCase
     assert_equal image.image_data.file_url, images.first['url']
   end
 
+  test "appends detailed document type information about an edition" do
+    news = create(:news_article)
+    publication = create(:publication)
+    speech = create(:speech)
+    corpinfo = create(:corporate_information_page)
+
+    newsresult = DocumentExportPresenter.new(news.document).as_json
+    doctype_data = newsresult[:editions].first[:document_type_information]
+    assert_equal news.news_article_type.as_json, doctype_data
+
+    pubresult = DocumentExportPresenter.new(publication.document).as_json
+    doctype_data = pubresult[:editions].first[:document_type_information]
+    assert_equal publication.publication_type.as_json, doctype_data
+
+    speechresult = DocumentExportPresenter.new(speech.document).as_json
+    doctype_data = speechresult[:editions].first[:document_type_information]
+    assert_equal speech.speech_type.as_json, doctype_data
+
+    corpresult = DocumentExportPresenter.new(corpinfo.document).as_json
+    doctype_data = corpresult[:editions].first[:document_type_information]
+    assert_equal corpinfo.corporate_information_page_type.as_json, doctype_data
+  end
+
   test "returns government information about an edition" do
+    create(:current_government)
```

Comment:
Is this meant to change?

Reply:
Yes. It appears creating an edition doesn't create any form of government for it in the backend, and there appears to be no means to do so built into the edition class.

Previously this test was "passing" only because we were comparing nil to nil - no government was assigned, rather than actually comparing the government data. Creating a government at the start of the test in this way allows for the test to run as intended, and matches patterns I've seen elsewhere (typically in a before step).

Reply:
@kevindew to clarify, a separate commit on this PR? Easy enough.

Reply:
@Tetrino yup, exactly that.